File: Week 1/3. mario more/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;

    // Prompt user for height
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Print pyramid
    for (int i = 0; i < height; i++)
    {
        // Print left pyramid
        for (int j = 0; j < height - i - 1; j++)
        {
            printf(" ");
        }
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }

        // Print gap
        printf("  ");

        // Print right pyramid
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }

        // Print newline
        printf("\n");
    }
    return 0;
}

File: Week 6/3. sentimental-mario-more/mario.py
# TODO
from cs50 import get_int

# get input from user until input is correct
while True:
    try:
        height = get_int("Height: ")
        if (height >= 1) and (height <= 8):
            break
    except:
        print("", end="")

for row in range(height):
    for space in range(height - row - 1, 0, -1):
        print(" ", end="")
    # print the left hashes
    for left_hash in range(row + 1):
        print("#", end="")
    # Print the middle seperation spaces
    print("  ", end="")
    # print the right hashes
    for right_hash in range(row + 1):
        print("#", end="")
    print()